proposals/stdlib/locale-agnostic-string-conversions.md

```diff
@@ -0,0 +1,159 @@
+# Locale-agnostic text processing by default
+
+* **Type**: Standard Library API proposal
+* **Author**: Abduqodiri Qurbonzoda
+* **Status**: Submitted
+* **Prototype**: Implemented
+
+## Summary
+
+Make text processing API locale-agnostic by default.
+
+## Current API review
+
+Currently, standard library provides following Common extension functions: 
+* `fun String.toLowerCase(): String`
+* `fun String.toUpperCase(): String`
+* `fun String.capitalize(): String`
+* `fun String.decapitalize(): String`
+
+The functions above convert letters of the receiver `String` using the rules of the default locale.
+
+Kotlin/JVM additionally provides overloads to specify the locale to be used:
+* `fun String.toLowerCase(locale: Locale): String`
+* `fun String.toUpperCase(locale: Locale): String`
+* `fun String.capitalize(locale: Locale): String`
+* `fun String.decapitalize(locale: Locale): String`
+
+`Char` has following locale-agnostic extension functions:
+* `fun Char.toLowerCase(): Char` - Common
+* `fun Char.toUpperCase(): Char` - Common
+* `fun Char.toTitleCase(): Char` - Kotlin/JVM
+
+## Motivation
+
+Our researches show that people often use locale-sensitive functions mentioned above without realizing the fact that their code behaves differently 
+in different geographies/platform locale settings. To combat the issue we would like to deprecate current API and introduce new locale-agnostic functions.
```

Comment:
It's worth to mention why is it error-prone to use default locale.
E.g. an existing bug, note from Java's `toLowerCase` documentation about its gotchas and so on.

Then the motivation of this proposal will be more understandable

Reply:
Added more details to the motivation part.